packages/cli/src/swagger/specGenerator3.ts

```diff
@@ -298,12 +298,17 @@ export class SpecGenerator3 extends SpecGenerator {
       };
 
       if (res.schema && !isVoidType(res.schema)) {
-        const produces = res.produces || defaultProduces || DEFAULT_RESPONSE_MEDIA_TYPE;
-        swaggerResponses[res.name].content = {
-          [produces]: {
-            schema: this.getSwaggerType(res.schema),
-          } as Swagger.Schema3,
-        };
+        swaggerResponses[res.name].content = {};
+        const produces: string[] = res.produces || defaultProduces || [DEFAULT_RESPONSE_MEDIA_TYPE];
+        for (const p of produces) {
+          const { content } = swaggerResponses[res.name];
+          swaggerResponses[res.name].content = {
+            ...content,
```

Comment:
normally I would do it simpler but this is the only working way where typescript compiler didn't complain